### PR TITLE
Add override field to rabbitmq controller

### PR DESCRIPTION
**Pull request by euclidgame**

### CODE REVIEW

File: src/deps_hack/src/lib.rs [hunk 1]
```diff
@@ -137,18 +139,47 @@ pub fn default_storage() -> k8s_openapi::apimachinery::pkg::api::resource::Quant
...
+    #[serde(rename = "storageClassName", default)]
     pub storage_class_name: Option<String>,
     #[serde(default = "default_storage")]
     pub storage: k8s_openapi::apimachinery::pkg::api::resource::Quantity,
 }
 
+/// Provides the ability to override the generated manifest of several child resources.
+#[derive(Default, Debug, Clone, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct RabbitmqClusterOverrideSpec {
+	/// Override configuration for the RabbitMQ StatefulSet.
```
Comment (marshtompsxd) [inline src/deps_hack/src/lib.rs:153]: indent

File: src/deps_hack/src/lib.rs [hunk 2]
```diff
@@ -137,18 +139,47 @@ pub fn default_storage() -> k8s_openapi::apimachinery::pkg::api::resource::Quant
...
+/// Allows for the manifest of the created StatefulSet to be overwritten with custom configuration.
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct RabbitmqOverrideStatefulSet {
+	pub spec: Option<RabbitmqOverrideStatefulSetSpec>,
+}
+
+/// StatefulSetSpec contains a subset of the fields included in k8s.io/api/apps/v1.StatefulSetSpec.
+/// Every field is made optional.
+#[derive(Debug, Clone, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct RabbitmqOverrideStatefulSetSpec {
```
Comment (marshtompsxd) [inline src/deps_hack/src/lib.rs:167]: I don't think we have to follow the same hierarchy as the official rabbitmq controller. We can promote `pod_management_policy` and `persistent_volume_claim_retention_policy` to the top level without having these middle layers.

Comment (euclidgame) [reply src/deps_hack/src/lib.rs:167]: What’s the benefit?

Comment (marshtompsxd) [reply src/deps_hack/src/lib.rs:167]: keep the interface simple

**Pull request merged**